src/com/magento/idea/magento2plugin/actions/generation/generator/DataModelGenerator.java

```diff
@@ -0,0 +1,141 @@
+    private final ValidatorBundle validatorBundle;
+    private final CommonBundle commonBundle;
+
+    /**
+     * Constructor.
+     */
+    public DataModelGenerator(final Project project, final DataModelData modelData) {
+        super(project);
+
+        this.project = project;
+        this.modelData = modelData;
+        this.directoryGenerator = DirectoryGenerator.getInstance();
+        this.fileFromTemplateGenerator = FileFromTemplateGenerator.getInstance(project);
+        this.getFirstClassOfFile = GetFirstClassOfFile.getInstance();
+        this.validatorBundle = new ValidatorBundle();
+        this.commonBundle = new CommonBundle();
+    }
+
+    @Override
+    public PsiFile generate(final String actionName) {
+        final PsiFile[] files = new PsiFile[1];
+
+        WriteCommandAction.runWriteCommandAction(project, () -> {
+            PhpClass model = GetPhpClassByFQN.getInstance(project).execute(
+                    modelData.getFQN()
+            );
+
+            if (model == null) {
+                model = createModel(actionName);
+
+                if (model == null) {
+                    final String errorMessage = this.validatorBundle.message(
+                            "validator.file.cantBeCreated",
+                            "Data Model"
+                    );
+                    JOptionPane.showMessageDialog(
+                            null,
+                            errorMessage,
+                            commonBundle.message("common.error"),
+                            JOptionPane.ERROR_MESSAGE
+                    );
+                } else {
+                    files[0] = model.getContainingFile();
+                }
+            } else {
+                final String errorMessage = this.validatorBundle.message(
+                        "validator.file.alreadyExists",
+                        "Data Model"
+                );
+                JOptionPane.showMessageDialog(
+                        null,
+                        errorMessage,
+                        commonBundle.message("common.error"),
+                        JOptionPane.ERROR_MESSAGE
+                );
+            }
+        });
+
+        return files[0];
+    }
+
+    @Override
+    protected void fillAttributes(final Properties attributes) {
+        final List<String> uses = getUses();
+        attributes.setProperty("NAMESPACE", modelData.getNamespace());
+        attributes.setProperty("USES", PhpClassGeneratorUtil.formatUses(uses));
+        attributes.setProperty("NAME", modelData.getName());
+        attributes.setProperty(
+                "EXTENDS",
+                PhpClassGeneratorUtil.getNameFromFqn(DataModel.DATA_OBJECT)
+        );
+        attributes.setProperty(
+                "IMPLEMENTS",
+                PhpClassGeneratorUtil.getNameFromFqn(modelData.getInterfaceFQN())
+        );
+        attributes.setProperty("PROPERTIES", modelData.getProperties());
```

Comment:
Convert an array of objects to string here.

Reply:
Fixed this, thank you!